### DISCUSSION TITLE
Nimib Speaking Hours - April 1st 10:30 CEST!

### DISCUSSION THREAD

**Discussion by pietroppeter** [pietroppeter/nimib#200] [Nimib speaking hours]
**April 1st 10:30-12 CEST** check the calendar link to add a reminder in your calendar (and in your timezone)! https://calndr.link/event/MXdGAsu8ng or directly https://meet.jit.si/NimibSpeakingHours

## Join us and say hi! 👋 

_(above edited to reflect discussion below)_ 
Did not have time during March to do much but we could still try and host and speaking hours next weekend, what do you think @HugoGranstrom ?

**Comment by HugoGranstrom** [comment]
Me neither, exactly 0 work done by me. But i guess this leaves more room for discussing new stuff ;) So I'm on! Saturday or Sunday shouldn't matter if we do it before/at lunch.

**Reply by pietroppeter** [reply]
> Me neither, exactly 0 work done by me.

Well that's exactly what we should do, an amount of work greater OR EQUAL to zero! 😆

For the agenda I was thinking maybe we could try to do the move to nimib org during the meeting? Ambitious but in this way if we mess up we mess up together...

I will align on what are my better times (likely better Sunday than Saturday) and plan the event. Everybody reading is invited to participate as usual, but I would like to mention specifically @dlesnoff and @neroist, our newest contributors. In case you have some time constraints and you are considering participating let us know and we will try to accomodate. We just chat about the progress (and later we post our notes in these discussion) and try to advance a bit the work.

**Reply by HugoGranstrom** [reply]
Moving repos sounds like a good plan. I would also like to discuss Nimibex and its potential structure.

**Reply by pietroppeter** [reply]
We should also find some whale related image for nimib-org repo avatar

**Comment by dlesnoff** [comment]
What kind of image? Would some image like the one in my allblocks PR fit?

Le mar. 28 mars 2023, 14:49, Pietro Peterlongo ***@***.***> a
écrit :

**Reply by pietroppeter** [reply]
haven't actually decided if it should be a photo (and the allblocks photo could be a good candidate, we might need to check the rights) or something more graphic (like the svg of emoji). it is up for discussion! :)

**Comment by pietroppeter** [comment]
Bit late but the proposal is Saturday 10.30-12 (up to 12.30) CET (April 1st but it is not a joke). How does that sound @HugoGranstrom ?

**Reply by HugoGranstrom** [reply]
Sounds good to me 👌 And just to be sure, you changed to summer time last week as well, right? 😅

**Reply by pietroppeter** [reply]
> Sounds good to me 👌 And just to be sure, you changed to summer time last week as well, right? 😅

Yep! ☀️

**Reply by HugoGranstrom** [reply]
Nice! 🚀 Would be fun if we had different times 🤣

**Reply by HugoGranstrom** [reply]
So to be more specific if we get any other guests, with CET we mean CEST (Central European Summer Time, UTC+2)? And not CET (UTC+1)?

**Reply by pietroppeter** [reply]
You are right! I have to prepare the calendar event...

**Reply by pietroppeter** [reply]
updated and added calendar event

**Comment by pietroppeter** [comment]
I am not sure I like the name nimib-org and I would like us to think about alternatives: nimib-team, nimib-docs, nimib.nim, ... let's discuss about this also today @HugoGranstrom !

**Reply by pietroppeter** [reply]
nb-team, nb-docs, ...

**Reply by HugoGranstrom** [reply]
nimib-land
nimib-party
nimib-home
nimib-dev
nimib-repos
nimib-code
nimib-nim
nimib-umbrella

**Reply by pietroppeter** [reply]
(from speaking hours...)
final candidates are:
- nimib-land
- nimib-home
- nimib-dev
and the winner is: Nimib-land! 👏 🐳 🌍

**Comment by pietroppeter** [comment]
Notes from meeting:
- what happened last month
  - nimitheme! both very happy about it!
  - https://github.com/pietroppeter/nimib/releases/tag/v0.3.9
    - we did only PR reviews, great contributions!
  - nblog basic theme / SSG discussion
    - discuss how to have changes to partial survive nim -> json -> html (example mermaid diagram)
      - option A: a renderProc that changes partial (generic in nim)
      - option B: headBlocks and in general blocks for different template parts
        - Pietro will proceed with this Option
    - custom render proc? load libraries dynamically
      - also needed when we want to work with nimja in the future
      - inspiration from this comes from **ivy: this is the way!**
    - stuff that will need to be ported to nimib
      - nbToJson
      - nbJsonToHtml
      - blocks everywhere approach (yet to be implemented in nblog)
      - template dir customizable in nimib.toml?
    - binary/library "compiled functionalities later when working on ivy (neve, refs "snow" italian but also wall-e)
    - climate is pretty good for SSG and could be adopted by nimib ecosystem
- nimib-org
  - name of nimib org changed to Nimib-land
- nimibex: https://github.com/nimib-org/nimibex/issues/1
- next?

**Comment by dlesnoff** [comment]
Hello,

Have you planned when April's meeting is going to be? I would like to
participate so that I can understand how to contribute most effectively to
the Nimib repository.

In particular, I would like to know whether the Nimib repository is going
to move to Nimib-land and how.

I wish you all the best,
Dimitri Lesnoff

Le sam. 1 avr. 2023 à 17:08, Pietro Peterlongo ***@***.***> a
écrit :

**Reply by pietroppeter** [reply]
Hi Dimitri is nice to hear from you! Unfortunately this past month has been a very busy period for me and it will continue being so for the next 2-3 weeks so I am afraid the next moment I will be available for speaking hours will be in the second half of May. I can confirm what @HugoGranstrom has said regarding nimib-land.

**Comment by HugoGranstrom** [comment]
We haven't planned it yet, but it's very nice to hear that you want to participate :smile: I don't have time this current weekend, but next weekend I should have time before lunch around 10am (CET summer time). Would that time work for you?

All nimib related repos will move to nimib-land, we just haven't had time left in the previous speaking hours to actually perform the moves.

**Comment by dlesnoff** [comment]
Hello,

I am unsure about this Saturday since I will be on the road.

> I will be available for speaking hours will be in the second half of May.

Nothing is urgent so I will wait until then.

Best regards,
@dlesnoff
